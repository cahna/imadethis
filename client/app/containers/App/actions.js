/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_USER_THREADS,
  LOAD_USER_THREADS_SUCCESS,
  LOAD_USER_THREADS_ERROR,
} from './constants';

/**
 * Load threads for a user, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_USER_THREADS
 */
export function loadUserThreads() {
  return {
    type: LOAD_USER_THREADS,
  };
}

/**
 * Dispatched when the threads for a user are loaded by the request saga
 *
 * @param  {array} threads The repository data
 * @param  {string} username The current username
 *
 * @return {object}      An action object with a type of LOAD_USER_THREADS_SUCCESS passing the threads
 */
export function userThreadsLoaded(threads, username) {
  return {
    type: LOAD_USER_THREADS_SUCCESS,
    threads,
    username,
  };
}

/**
 * Dispatched when loading the threads for a user fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_USER_THREADS_ERROR passing the error
 */
export function userThreadLoadingError(error) {
  return {
    type: LOAD_USER_THREADS_ERROR,
    error,
  };
}
