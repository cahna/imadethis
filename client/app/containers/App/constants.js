/**
 * Action/saga names
 */
export const REQUEST_LOGOUT = 'imadethis/App/REQUEST_LOGOUT';
export const LOGOUT_SUCCESS = 'imadethis/App/LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'imadethis/App/LOGOUT_FAILED';
export const USER_LOGGED_IN = 'imadethis/App/USER_LOGGED_IN';
export const GET_ACTIVE_USER = 'imadethis/App/GET_ACTIVE_USER';
export const LOADING_ACTIVE_USER = 'imadethis/App/LOADING_ACTIVE_USER';
export const ACTIVE_USER_LOADED = 'imadethis/App/ACTIVE_USER_LOADED';

/**
 * API routes
 */
export const API_LOGIN = '/api/auth/login';
export const API_LOGOUT = '/api/auth/logout';
export const API_ACTIVE_USER = '/api/auth/active_user';

/**
 * Configuration / other
 */
export const LOCAL_TOKEN_NAME = 'accessToken';
