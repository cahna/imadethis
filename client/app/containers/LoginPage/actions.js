import {
  USERNAME_CHANGED,
  PASSWORD_CHANGED,
  REQUEST_LOGIN,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  RESET_LOGIN_PAGE,
} from './constants';

export const changeUsername = username => ({
  type: USERNAME_CHANGED,
  payload: { username },
});

export const changePassword = password => ({
  type: PASSWORD_CHANGED,
  payload: { password },
});

export const submitLogin = () => ({ type: REQUEST_LOGIN });

export const loginSuccess = accessToken => ({
  type: LOGIN_SUCCESS,
  payload: { accessToken },
});

export const loginFailure = () => ({ type: LOGIN_FAILURE });

export const resetLoginPage = () => ({ type: RESET_LOGIN_PAGE });
