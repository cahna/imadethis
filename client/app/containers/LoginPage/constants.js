/**
 * LoginPage constants
 */

export const REQUEST_LOGIN = 'imadethis/LoginPage/REQUEST_LOGIN';
export const LOGIN_SUCCESS = 'imadethis/LoginPage/LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'imadethis/LoginPage/LOGIN_FAILURE';
export const USERNAME_CHANGED = 'imadethis/LoginPage/USERNAME_CHANGED';
export const PASSWORD_CHANGED = 'imadethis/LoginPage/PASSWORD_CHANGED';

export const API_LOGIN_URL = '/api/auth/login';
