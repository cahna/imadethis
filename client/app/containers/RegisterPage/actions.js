import {
  USERNAME_CHANGED,
  PASSWORD_CHANGED,
  CONFIRM_PASSWORD_CHANGED,
  REQUEST_REGISTER,
  REGISTER_SUCCESS,
  REGISTER_FAILURE,
  RESET_REGISTER_PAGE,
} from './constants';

export const changeUsername = (username) => ({
  type: USERNAME_CHANGED,
  payload: { username },
});

export const changePassword = (password) => ({
  type: PASSWORD_CHANGED,
  payload: { password },
});

export const changeConfirmPassword = (confirmPassword) => ({
  type: CONFIRM_PASSWORD_CHANGED,
  payload: { confirmPassword },
});

export const submitRegister = () => ({ type: REQUEST_REGISTER });

export const registerSuccess = () => ({ type: REGISTER_SUCCESS });

export const registerFailure = (error = '') => ({
  type: REGISTER_FAILURE,
  payload: { error },
});

export const resetRegisterPage = () => ({ type: RESET_REGISTER_PAGE });
