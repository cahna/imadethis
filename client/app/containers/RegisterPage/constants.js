/**
 * RegisterPage constants
 */

export const REQUEST_REGISTER = 'imadethis/RegisterPage/REQUEST_REGISTER';
export const REGISTER_SUCCESS = 'imadethis/RegisterPage/REGISTER_SUCCESS';
export const REGISTER_FAILURE = 'imadethis/RegisterPage/REGISTER_FAILURE';
export const USERNAME_CHANGED = 'imadethis/RegisterPage/USERNAME_CHANGED';
export const PASSWORD_CHANGED = 'imadethis/RegisterPage/PASSWORD_CHANGED';
export const CONFIRM_PASSWORD_CHANGED =
  'imadethis/RegisterPage/CONFIRM_PASSWORD_CHANGED';
export const REGISTER_FORM_LOADING =
  'imadethis/RegisterPage/REGISTER_FORM_LOADING';
