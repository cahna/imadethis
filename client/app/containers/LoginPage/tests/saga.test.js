import { put, takeLatest } from 'redux-saga/effects';

import { REQUEST_LOGIN } from '../constants';
import { loginSuccess, loginFailure } from '../actions';
import loginPageSaga, { submitLogin } from '../saga';

const accessToken = '_JWT_';

/* eslint-disable redux-saga/yield-effects */
describe('loginPageSaga', () => {
  const mainSaga = loginPageSaga();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = mainSaga.next().value;

    expect(takeLatestDescriptor).toEqual(
      takeLatest(REQUEST_LOGIN, submitLogin),
    );
  });
});

describe('submitLogin saga generator', () => {
  let submitLoginGenerator;

  beforeEach(() => {
    submitLoginGenerator = submitLogin();

    const selectUsernameDescriptor = submitLoginGenerator.next().value;
    expect(selectUsernameDescriptor).toMatchSnapshot();

    const selectPasswordDescriptor = submitLoginGenerator.next().value;
    expect(selectPasswordDescriptor).toMatchSnapshot();
  });

  afterEach(() => {
    expect(submitLoginGenerator.next().done).toEqual(true);
  });

  it('should dispatch the loginSuccess action if request is successfull', () => {
    const callApiDescriptor = submitLoginGenerator.next({
      access_token: accessToken,
    }).value;
    expect(callApiDescriptor).toMatchSnapshot();

    const putDescriptor = submitLoginGenerator.next().value;
    expect(putDescriptor).toEqual(put(loginSuccess(accessToken)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const callApiDescriptor = submitLoginGenerator.throw().value;
    expect(callApiDescriptor).toMatchSnapshot();

    const putDescriptor = submitLoginGenerator.next().value;
    expect(putDescriptor).toEqual(put(loginFailure()));
  });
});
