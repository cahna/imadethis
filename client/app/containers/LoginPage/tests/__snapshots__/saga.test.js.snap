// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitLogin saga generator should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`submitLogin saga generator should call the repoLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`submitLogin saga generator should call the repoLoadingError action if the response errors 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "imadethis/LoginPage/LOGIN_FAILURE",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`submitLogin saga generator should dispatch the loginSuccess action if request is successfull 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`submitLogin saga generator should dispatch the loginSuccess action if request is successfull 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`submitLogin saga generator should dispatch the loginSuccess action if request is successfull 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/auth/login",
      Object {
        "body": "{\\"password\\":{\\"access_token\\":\\"_JWT_\\"}}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
